crates/context_aware_config/src/api/context/helpers.rs

```diff
@@ -154,11 +154,11 @@ fn get_functions_map(
 
 pub fn validate_value_with_function(
     _fun_name: &str,
-    function: &str,
+    function: FunctionCode,
```

Comment:
also, shouldn't the ref be passed along  ? 
```suggestion
    function: &FunctionCode,
```